e2e/integrations/edit/edit.po.ts

```diff
@@ -209,52 +209,249 @@ export class IntegrationConfigureLogStepPage extends IntegrationConfigureStepPag
+      this.setValue(rule.getValue());
+    }
+
+    return this.setPredicate(this.predicate);
   }
 
   validate(): P<any> {
     log.debug(`validating configuration page`);
-    return this.getFilterDefinitioTextArea().isPresent();
+
+    const predicatPresent = this.getPredicateSelect().isPresent();
+    const pathPresent = this.getPathInput().isPresent();
+    const valuePresent = this.getValueInput().isPresent();
+    const opPresent = this.getOpSelect().isPresent();
+
+    return (predicatPresent && pathPresent && valuePresent && opPresent);
   }
 
-  initialize(): P<any> {
-    return this.getFilterDefinitioTextAreaValue().then((function(text) {
-      this.setParameter(text);
-    }).bind(this));
+  async initialize(): P<any> {
+    const predicateSelectValue = await this.getPredicateSelectValue();
+    const pathInputValue = await this.getPathInputValue();
+    const opSelectValue = await this.getOpSelectValue();
+    const valueInputValue = await this.getValueInputValue();
+
+    const parameter = predicateSelectValue + ', ' + pathInputValue + ', ' + opSelectValue + ', ' + valueInputValue;
+
+    return this.setParameter(parameter);
   }
 
-  setFilterCondition(filterCondition: string): P<any> {
-    log.info(`setting integration filter step condition to ${filterCondition}`);
-    return this.rootElement().$(IntegrationConfigureFilterStepPage.filterSelector).sendKeys(filterCondition);
+  async addRule(ruleString: string): P<any> {
+    const ruleStringArray = ruleString.split(', ');
+    const op = ruleStringArray[1];
+    const basicFilterRule = new BasicFilterRule(ruleStringArray[0], BasicFilterOps[op], ruleStringArray[2]);
+
+    const addRuleLink = await this.rootElement().$(IntegrationConfigureBasicFilterStepPage.addRuleSelector);
+    await addRuleLink.click();
+
+    this.setLatestPathInput(basicFilterRule.getPath());
+    this.setLatestOpSelect(basicFilterRule.getOp());
+    this.setLatestValueInput(basicFilterRule.getValue());
+
+    this.ruleArray.push(basicFilterRule);
   }
 
   setParameter(filterCondition: string): void {
     this.filterCondition = filterCondition;
   }
 
-  getFilterDefinitioTextArea(): ElementFinder {
-    log.debug(`searching filter definition text area`);
-    return this.rootElement().$(IntegrationConfigureFilterStepPage.filterSelector);
+  setPredicate(predicate: number): P<any> {
+    log.info(`setting basic filter step predicate to option number ${predicate}`);
+    const predicateOptions = this.rootElement().all(by.css(IntegrationConfigureBasicFilterStepPage.predicateOptionSelector));
+    return predicateOptions.then((options) => {
+      options[this.predicate].click();
+    });
+  }
+
+  setOp(op: number): P<any> {
+    log.info(`setting basic filter step predicate to option number ${op}`);
+    const opOptions = this.rootElement().all(by.css(IntegrationConfigureBasicFilterStepPage.opOptionSelector));
+    return opOptions.then((options) => {
+      options[op].click();
+    });
+  }
+
```

Comment:
`op` is index? I'd probably prefer operation name or some abbreviation rather than index. Purely due to improved readability of code :)